test/OpenTelemetry.Tests/Trace/TracerProviderBuilderExtensionsTest.cs

```diff
@@ -133,6 +133,75 @@ public void ServiceLifecycleAvailableToSDKBuilderTest()
         Assert.True(myInstrumentation.Disposed);
     }
 
+    [Fact]
+    public void AddProcessorTest()
+    {
+        List<MyProcessor> processors = new();
+
+        using (var provider = Sdk.CreateTracerProviderBuilder()
+            .AddProcessor<MyProcessor>()
+            .AddProcessor(new MyProcessor()
+            {
+                PipelineWeight = 10_000,
+            })
+            .AddProcessor(new MyProcessor()
+            {
+                PipelineWeight = -10_000,
+            })
+            .AddProcessor(sp => new MyProcessor()
+            {
+                PipelineWeight = int.MaxValue,
+            })
+            .AddProcessor(new MyProcessor()
+            {
+                PipelineWeight = int.MinValue,
+            })
+            .SetErrorStatusOnException() // Forced to be first processor
+            .Build() as TracerProviderSdk)
+        {
+            Assert.NotNull(provider);
+            Assert.NotNull(provider.Processor);
+
+            var compositeProcessor = provider.Processor as CompositeProcessor<Activity>;
+
+            Assert.NotNull(compositeProcessor);
+
+            bool isFirstProcessor = true;
+            var lastWeight = int.MinValue;
+            var current = compositeProcessor.Head;
+            while (current != null)
+            {
+                if (isFirstProcessor)
```

Comment:
nit: We could avoid condition here by keeping the first iteration out of while loop.